woocommerce/google-listings-and-ads
# Consolidate accounts onboarding: Connect to existing Google Ads account via the combo card

### Changes proposed in this Pull Request:



Closes #2596  .

_Replace this with a good description of your changes & reasoning._


### Screenshots:


<img width="1095" alt="Screenshot 2024-10-04 at 10 37 36 PM" src="https://github.com/user-attachments/assets/ca42481b-88ea-4637-bc27-fb7b0db29bd6">


### Detailed test instructions:

There are mainly two scenarios to test for this issue.

### Scenario 1
Both Google Ads and MC accounts are present for the Google account.

### Scenario 2
Google Ads exists, but no MC account is associated with the Google account.

Install [tweak extension](https://tweak-extension.com/docs/intro) in chromw browser to test both the scenarios.

#### Rules for scenario 1

**Rule 1:**

URL: https://domain-name.ngrok.app/index.php?rest_route=%2Fwc%2Fgla%2Fmc%2Fconnection&_locale=user  
Method: GET  
Delay: 2000ms  
Response: 
```js
{
  "id": 1234567,
  "currency": "USD",
  "symbol": "$",
  "status": "connected",
  "step": "",
}
```

**Rule 2:**

URL: https://domain-name.ngrok.app/index.php?rest_route=%2Fwc%2Fgla%2Fads%2Faccounts&_locale=user  
Method: GET  
Delay: 2000ms  
Response: 
```js
[
  {
    "id": 1234567,
    "subaccount": true,
    "name": "google-listings-and-ads",
    "domain": "https://domain-name.ngrok.app"
  }
]
```

#### Rules for scenario 2

**Rule 1:**

URL: https://domain-name.ngrok.app/index.php?rest_route=%2Fwc%2Fgla%2Fmc%2Fconnection&_locale=user  
Method: GET  
Delay: 2000ms  
Response: 
```js
{
  "id": 0,
  "currency": "USD",
  "symbol": "$",
  "status": "disconnected",
  "step": "",
}
```

**Rule 2:**

URL: https://domain-name.ngrok.app/index.php?rest_route=%2Fwc%2Fgla%2Fads%2Faccounts&_locale=user  
Method: GET  
Delay: 2000ms  
Response: 
```js
[]
```

### Steps to test

1. Only enable the scenario related rules in extension. Enable extension by clicking play button.

2. Go to Step 1 of the onboarding, make sure to connect with Google account which already have existing ads accounts associated.

3. Once Google account is connected, you should see `Connect to existing Google Ads account` card with the ads accounts in dropdown.

4. If there is only one account, the dropdown won't be able to expand.

5. Clicking the connect button will trigger connection process and will connect to that ads account. You should be able to see ads account ID in card description once connected successfully.

6. For scenario 2, you should be able to see that merchant account is being created. It will always be in creation status because we are mocking that state via extension.

7. Once the ads account is connected, the dropdown won't be able to expand. It will show the `Connected` label beside dropdown.

8. If ads account is connected, text in footer should be `Or, connect to a different Google Ads account`.


### Additional details:


### Changelog entry
